enterprise/frontend/src/embedding-sdk/components/private/PublicComponentWrapper/PublicComponentWrapper.unit.spec.tsx

```diff
@@ -0,0 +1,65 @@
+import { renderWithProviders, screen } from "__support__/ui";
+import { SDK_REDUCERS } from "embedding-sdk/store";
+import type { LoginStatus } from "embedding-sdk/store/types";
+import {
+  createMockLoginStatusState,
+  createMockSdkState,
+} from "embedding-sdk/test/mocks/state";
+import { createMockState } from "metabase-types/store/mocks";
+
+import { PublicComponentWrapper } from "./PublicComponentWrapper";
+
+const setup = (status: LoginStatus = { status: "uninitialized" }) => {
+  const state = createMockState({
+    sdk: createMockSdkState({
+      loginStatus: createMockLoginStatusState(status),
+    }),
+  });
+
+  renderWithProviders(
+    <PublicComponentWrapper>
+      <div>My component</div>
+    </PublicComponentWrapper>,
+    {
+      storeInitialState: state,
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```

Comment:
Like below - the type is a bit wild and I don't think it's worth going down the rabbit hole to solve this right now. I think once we get types for the store a little more settled we can come back to this

Reply:
Also I intentionally didn't use `mode:sdk` here - when using `renderWithProviders` with `mode:sdk`, we're setting up the whole SDK instance with `use-init-data` and the like, which modify the loginStatus in its own processes, which make test cases a lot harder to anticipate. This unit test aims to isolate the component and test its ability to display all of the loginStatus states.

Reply:
Much nicer. Thanks!